packages/build/tests/core/snapshots/tests.js.md

```diff
@@ -2249,3 +2249,105 @@ Generated by [AVA](https://avajs.dev).
+      Netlify Build                                                 ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    > Version␊
+      @netlify/build 1.0.0␊
+    ␊
+    > Flags␊
+      debug: true␊
+      featureFlags:␊
+        - buildbot_es_modules_esbuild␊
+      repositoryRoot: packages/build/tests/core/fixtures/functions_es_modules␊
+      testOpts:␊
+        pluginsListUrl: test␊
+        silentLingeringProcesses: true␊
+    ␊
+    > Current directory␊
+      packages/build/tests/core/fixtures/functions_es_modules␊
+    ␊
+    > Config file␊
+      packages/build/tests/core/fixtures/functions_es_modules/netlify.toml␊
+    ␊
+    > Resolved config␊
+      build:␊
+        publish: packages/build/tests/core/fixtures/functions_es_modules␊
+        publishOrigin: default␊
+      functionsDirectory: packages/build/tests/core/fixtures/functions_es_modules/functions␊
+    ␊
+    > Context␊
+      production␊
+    ␊
+    ────────────────────────────────────────────────────────────────␊
+      1. Functions bundling                                         ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    Packaging Functions from functions directory:␊
+     - function_with_es_module.js␊
+    ␊
+    ␊
+    (Functions bundling completed in 1ms)␊
+    ␊
+    ────────────────────────────────────────────────────────────────␊
+      Netlify Build Complete                                        ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    (Netlify Build completed in 1ms)`
+
+## Fails build for ES module function if feature flag is off
+
+> Snapshot 1
+
+    `␊
+    ────────────────────────────────────────────────────────────────␊
+      Netlify Build                                                 ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    > Version␊
+      @netlify/build 1.0.0␊
+    ␊
+    > Flags␊
+      debug: true␊
+      repositoryRoot: packages/build/tests/core/fixtures/functions_es_modules␊
+      testOpts:␊
+        pluginsListUrl: test␊
+        silentLingeringProcesses: true␊
+    ␊
+    > Current directory␊
+      packages/build/tests/core/fixtures/functions_es_modules␊
+    ␊
+    > Config file␊
+      packages/build/tests/core/fixtures/functions_es_modules/netlify.toml␊
+    ␊
+    > Resolved config␊
+      build:␊
+        publish: packages/build/tests/core/fixtures/functions_es_modules␊
+        publishOrigin: default␊
+      functionsDirectory: packages/build/tests/core/fixtures/functions_es_modules/functions␊
+    ␊
+    > Context␊
+      production␊
+    ␊
+    ────────────────────────────────────────────────────────────────␊
+      1. Functions bundling                                         ␊
+    ────────────────────────────────────────────────────────────────␊
+    ␊
+    Packaging Functions from functions directory:␊
+     - function_with_es_module.js␊
```

Comment:
Shouldn't bundling fail in this instance?

Reply:
YUP! not really sure why it's not... threw up this draft to see if you or @eduardoboucas could help me figure out why it's not 😓

Reply:
The [debug mode](https://github.com/netlify/build/blob/main/packages/build/tests/README.md#debug-mode) might help you: combined with `test.only()`, this will allow you to run the test and see the build output. You can then add debugger or `console.log()` statements in `@netlify/build` or in `node_modules/@netlify/zip-it-and-ship-it` to debug what's going on.

Reply:
as per @eduardoboucas:

> the build wouldn’t fail, you should simply see a runtime error when accessing the function.

so I'll see if I can find a way to assert on the output

Reply:
@ehmicky and I agreed this is tested by zip-it-and-ship-it, so there's no need for a new test other than the two I've added to ensure the build passes with `true` and `false` flag values